### PR TITLE
Merge 48-node-serialization into main

### PR DESCRIPTION
**Pull request by originalmk**
This MR separates saving logic from the Node moving it to NodeStore. Node implements Into<NodeSnapshot> and such snapshots may be passed to NodeStore for saving. Whenever Node needs to be loaded from NodeStore user should firstly get a node snapshot by sending LoadNode message and then using NodeSnapshot in the NodeBuilder (from_snapshot method).

### CODE REVIEW

File: liberum_core/src/node/mod.rs [hunk 1]
```diff
@@ -220,65 +232,65 @@ impl Node {
...
+                response_sender: send,
+            })
+            .await?;
+
+        let id_str = liberum_core::file_id_to_str(id);
+
+        match recv.await {
+            Ok(Ok(_)) => Ok(id_str),
+            Ok(Err(e)) => Err(e.into()),
+            Err(e) => Err(e.into()),
```
Comment (philipandag) [inline liberum_core/src/node/mod.rs:292]: I hate those github diffs xD

File: liberum_core/src/node/store.rs [hunk 2]
```diff
@@ -27,12 +30,17 @@ pub enum NodeStoreError {
     NodeDoesNotExist { name: String },
     #[error("other error, name: {name}, err: {err}")]
     OtherError { name: String, err: anyhow::Error },
+    #[error("any error, err: {err}")]
+    Any {
```
Comment (philipandag) [inline liberum_core/src/node/store.rs:34]: That won't help with the error handling...

File: liberum_core/src/node/store.rs [hunk 3]
```diff
@@ -50,30 +58,76 @@ impl NodeStore {
...
+            );
+
+            return Err(NodeStoreError::OtherError {
+                name,
+                err: anyhow!("node_dir_path is not a directory"),
+            });
+        }
 
-        Ok(node)
+        let config_path = node_dir_path.join(Node::CONFIG_FILE_NAME);
```
Comment (philipandag) [inline liberum_core/src/node/store.rs:73]: Shouldn't those constants be moved to the store.rs module?

File: liberum_core/src/node/store.rs [hunk 4]
```diff
@@ -230,16 +284,19 @@ mod tests {
             .await
             .unwrap();
         let node_store = kameo::spawn(node_store);
-        let new_node = Node::builder()
-            .name("test_node".to_string())
-            .keypair(Keypair::generate_ed25519())
-            .build()
-            .unwrap();
+        let node_snapshot = NodeSnapshot {
```
Comment (philipandag) [inline liberum_core/src/node/store.rs:287]: Couldn't an alternative method to `build` be added to the builder, which could be used to build the snapshots? Most of the fields are the same.

**Review by philipandag — COMMENTED** [review]
Haven't found anything unacceptable. Let's just discuss the comments

File: liberum_core/src/node/store.rs (hunk 2 above)
Comment (originalmk) [reply liberum_core/src/node/store.rs]: Fixed.

File: liberum_core/src/node/store.rs (hunk 3 above)
Comment (originalmk) [reply liberum_core/src/node/store.rs]: Done.

File: liberum_core/src/node/store.rs (hunk 2 above)
Comment (originalmk) [reply liberum_core/src/node/store.rs]: https://github.com/Liberum-Neto/liberum-neto/commit/b8a0902ccf38d9d357c77bf6730cb027c9e6c7d4

File: liberum_core/src/node/store.rs (hunk 3 above)
Comment (originalmk) [reply liberum_core/src/node/store.rs]: https://github.com/Liberum-Neto/liberum-neto/commit/b8a0902ccf38d9d357c77bf6730cb027c9e6c7d4

**Review by philipandag — APPROVED** [review]
ok

**Pull request merged**